### PR TITLE
Debugging tutorial for Knative Eventing

### PR DESCRIPTION
**Pull request by Harwayne**
## Proposed Changes

- Debugging tutorial for Knative Eventing.
    - Needs more work, but is a starting place.

### CODE REVIEW

File: eventing/debugging/README.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
+# Debugging Knative Eventing
+
+This is an evolving document on how to debug a non-working Knative Eventing setup.
+## Audience
+
+This document is intended for people that are familiar with [Knative Eventing](../README.md)'s object model. You don't need to be an expert, but do need to know roughly how things fit together.
+
+## Version
+This document is works with [Eventing 0.3](https://github.com/knative/eventing/releases/tag/v0.3.0) and [Eventing Sources 0.3](https://github.com/knative/eventing-sources/releases/tag/v0.3.0).
```
Comment (vaikas-google) [inline eventing/debugging/README.md]: This document works

File: eventing/debugging/README.md [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+```
+
+Then we know there was a problem posting to `http://svc.knative-debug.svc.cluster.local/`.
+
+TODO Finish this section.
```
Comment (vaikas-google) [inline eventing/debugging/README.md]: add the future work on emitting k8s events for failures here and ship it.

File: eventing/debugging/README.md (hunk 1 above)
Comment (Harwayne) [reply eventing/debugging/README.md]: Done.

File: eventing/debugging/README.md (hunk 2 above)
Comment (Harwayne) [reply eventing/debugging/README.md]: Done.

**Comment by vaikas-google** [comment]
/approve

letting @nachocano lgtm it in case he has any feedback.

**Comment by vaikas-google** [comment]
/lgtm
I think the /lgtm needs to be on a line by itself for it to be accepted.

**Pull request merged**